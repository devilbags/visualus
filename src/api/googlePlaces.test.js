import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import GooglePlaces from './googlePlaces';

describe('Google Places API', () => {
  const url = 'https://maps.googleapis.com/maps/api/place/autocomplete/json';
  it('can fetch', async () => {
    expect.assertions(1);
    const mock = new MockAdapter(axios);
    mock.onGet(url).reply(200, {
      city: 'Boston'
    });
    const data = await GooglePlaces.fetchPlaces();
    expect(data.city).toEqual('Boston');
  });
  it('can handle error', async () => {
    const mock = new MockAdapter(axios);
    mock.onGet(url).networkError();
    await GooglePlaces.fetchPlaces().catch(error => {
      expect(error).toMatchSnapshot();
    });
  });
});
