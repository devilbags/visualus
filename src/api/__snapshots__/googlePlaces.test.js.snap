// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Places API can handle error 1`] = `[Error: Network Error]`;
